File: ProgTechGy/y2xwhj_pt_beadando/src/main/java/strategy_pattern/contexts/BillingContext.java
package strategy_pattern.contexts;

import strategy_pattern.interfaces.BillingStrategy;

public class BillingContext {
    private BillingStrategy strategy;

    public BillingContext(BillingStrategy strategy) {
        this.strategy = strategy;
    }

    public void setStrategy(BillingStrategy strategy) {
        this.strategy = strategy;
    }

    public double checkout(double amount) {
        return strategy.getFinalAmount(amount);
    }
    
}

File: ProgTechEa/MindenEgybe/BillingStrategy/BillingContext.cs
using BillingStrategy.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BillingStrategy
{
    internal class BillingContext
    {
        private IBillingStrategy _billingStrategy;

        public BillingContext(IBillingStrategy billingStrategy)
        {
            _billingStrategy = billingStrategy;
        }

        public void SetBillingStrategy(IBillingStrategy billingStrategy)
        {
            _billingStrategy = billingStrategy;
        }

        public double Checkout(double price)
        {
            return _billingStrategy.CreateBill(price);
        }
    }
}
